app/src/main/kotlin/com/exallium/todoapp/notedetail/NoteDetailPresenter.kt

```diff
@@ -0,0 +1,42 @@
+package com.exallium.todoapp.notedetail
+
+import com.exallium.todoapp.entities.Note
+import com.exallium.todoapp.mvp.BasePresenter
+import com.exallium.todoapp.screenbundle.ScreenBundleHelper
+import rx.Single
+import rx.SingleSubscriber
+import timber.log.Timber
+
+/**
+ * Presenter for Note Detail Screen
+ */
+class NoteDetailPresenter(val view: NoteDetailView, val model: NoteDetailModel,
+                          val screenBundleHelper: ScreenBundleHelper) : BasePresenter<NoteDetailView>(view) {
+    lateinit var note: Note
+    val noteId: String = screenBundleHelper.getNoteId(view.retrieveBundle())
+
+    override fun onViewCreated() {
+        if (view.isCached(noteId)) {
+            return
+        }
+        lookupNoteDetail(noteId)
+        view.setNoteData(note)
+    }
+
+    fun lookupNoteDetail(noteId: String) {
+        Timber.d("NOTE: " + noteId)
+        var noteSingle: Single<Note> = model.getNoteById(noteId)
+
+        var subscription = noteSingle.subscribe(object : SingleSubscriber<Note>() {
+            override fun onSuccess(t: Note) {
+                note = t
+            }
+            override fun onError(t: Throwable) {
+                Timber.d(t, "error looking up note with id " + noteId)
+                throw t
```

Comment:
done. 2 comments:

1) I was not able to get snackbar to show up anywhere I tried (using view.displaySnackbar(R.string.note_detail_screen_error).
2) Do you think it's ok to call it from Presenter and delegate what message to display to view since it is a view logic?
3) I would remove the displaySnackbar function that takes a String instead of StringResId so that all Strings can be internationalized thru Resources if necessary